### PR TITLE
Mention right associativity of the `--/2` operator in docs

### PR DESCRIPTION
**Pull request by tajchumber**
Just thought I would add a small mention to this as it has caused a few headaches to me ahah!

Happy to add more if needed or move it to a different place.

### CODE REVIEW

File: lib/elixir/lib/kernel.ex [hunk 1]
```diff
@@ -1384,6 +1384,11 @@ defmodule Kernel do
       iex> [1, 2, 3, 2, 1] -- [1, 2, 2]
       [3, 1]
 
+  It is important to note that `--/2` is right associative meaning:
```
Comment (eksperimental) [inline lib/elixir/lib/kernel.ex:1387]: I think to clarify the example it should also show how it is interpreted with parentheses.

File: lib/elixir/lib/kernel.ex [hunk 2]
```diff
@@ -1384,6 +1384,11 @@ defmodule Kernel do
...
+
+      iex> [1, 2, 3] -- [2] -- [3]
+      [1, 3]
+
```
Comment (josevalim) [inline lib/elixir/lib/kernel.ex:1391]:
```suggestion

  As it is equivalent to:

      iex> [1, 2, 3] -- ([2] -- [3])
      [1, 3]

```

**Comment by eksperimental** [comment]
For completeness we should add the same to ++/2

**Comment by josevalim** [comment]
It doesn't matter on `++` because the order of the operands do not matter there, at least in terms of correctness.

File: lib/elixir/lib/kernel.ex [hunk 3]
```diff
@@ -1384,6 +1384,16 @@ defmodule Kernel do
       iex> [1, 2, 3, 2, 1] -- [1, 2, 2]
       [3, 1]
 
+  It is important to note that `--/2` is right associative meaning:
```
Comment (eksperimental) [inline lib/elixir/lib/kernel.ex:1387]:
```suggestion
  It is important to note that the `--/2` operator has a *right to left* associativity:
```

Comment (josevalim) [reply lib/elixir/lib/kernel.ex:1387]: I personally find "right associative" clearer, which seems to be correct usage from the wikipedia page: https://en.wikipedia.org/wiki/Operator_associativity

Comment (josevalim) [reply lib/elixir/lib/kernel.ex:1387]:
```suggestion
The `--/2` operator is right associative, meaning:
```

**Pull request merged**